server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -172,6 +172,13 @@ public boolean isSet() {
             return Integer.valueOf(get()).equals(defaultValue) == false;
         }
 
+        /**
+         * Returns whether values are potentially ignored, either by an explicitly configured ignore_above or by the default value.
+         */
+        public boolean valuesPotentiallyIgnored() {
+            return get() != Integer.MAX_VALUE;
```

Comment:
[nit] can you leave a comment explaining why `Integer.MAX_VALUE` is used here? Otherwise its use clashes with the default value for ignore_above for non-logsdb indices, which is confusing.